identity-android/src/main/java/com/android/identity/android/mdoc/transport/DataTransport.java

```diff
@@ -245,31 +261,47 @@ protected void reportMessageReceived(@NonNull byte[] data) {
         final Listener listener = mListener;
         final Executor executor = mListenerExecutor;
         if (!mInhibitCallbacks && listener != null && executor != null) {
-            executor.execute(() -> listener.onMessageReceived());
+            executor.execute(() -> {
+                if (!mInhibitCallbacks) {
+                    listener.onMessageReceived();
+                }
+            });
         }
     }
 
     protected void reportMessageProgress(long progress, long max) {
         final TransmissionProgressListener listener = mProgressListener;
         final Executor executor = mProgressListenerExecutor;
         if (!mInhibitCallbacks && listener != null && executor != null) {
-            executor.execute(() -> listener.onProgressUpdate(progress, max));
+            executor.execute(() -> {
+                if (!mInhibitCallbacks) {
+                    listener.onProgressUpdate(progress, max);
+                }
+            });
         }
     }
 
     protected void reportTransportSpecificSessionTermination() {
         final Listener listener = mListener;
         final Executor executor = mListenerExecutor;
         if (!mInhibitCallbacks && listener != null && executor != null) {
```

Comment:
It is a little weird to check mInhibitCallback twice, rather than just in the execution  (it means we don't do callbacks if the boolean is false on initial call or when resolving the callback) Probably, that makes sense but it might be reasonable to only do this in the runnable.

Reply:
I agree. I guess I was trying to prematurely optimize not scheduling something on the executor but it's really not worth it and I agree a bit weird/confusing. I'll remove it.